test/quantization/core/experimental/test_linear.py

```diff
@@ -0,0 +1,40 @@
+# Owner(s): ["oncall: quantization"]
+
+import torch
+from torch.ao.quantization.experimental.linear import LinearAPoT
+import unittest
+
+class TestNonUniformObserver(unittest.TestCase):
+    """
+        Test linear_APoT_fn by calling forward method
+    """
+    def test_linear_APoT_fn(self):
```

Comment:
you should compare with an actual linear. This basically redoes the bit shift implementation as a multiply but there's more going on than just that which needs to be tested. A full end to end comparison with a float linear would help with that.